tests/components/ituran/test_config_flow.py

```diff
@@ -0,0 +1,311 @@
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ID_OR_PASSPORT: MOCK_CONFIG_DATA[CONF_ID_OR_PASSPORT],
+            CONF_PHONE_NUMBER: MOCK_CONFIG_DATA[CONF_PHONE_NUMBER],
+        },
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "invalid_auth"}
+
+    mock_ituran.request_otp.side_effect = None
+    result2 = await __do_successful_user_step(hass, result, mock_ituran)
+    await __do_successful_otp_step(hass, result2, mock_ituran)
+
+
+async def test_invalid_otp(
+    hass: HomeAssistant, mock_ituran: AsyncMock, mock_setup_entry: AsyncMock
+) -> None:
+    """Test invalid OTP configuration flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result2 = await __do_successful_user_step(hass, result, mock_ituran)
+
+    mock_ituran.authenticate.side_effect = IturanAuthError
+    result3 = await hass.config_entries.flow.async_configure(
+        result2["flow_id"],
+        user_input={
+            CONF_OTP: "123456",
+        },
+    )
+
+    assert result3["type"] is FlowResultType.FORM
+    assert result3["errors"] == {"base": "invalid_otp"}
+
+    mock_ituran.authenticate.side_effect = None
+    await __do_successful_otp_step(hass, result3, mock_ituran)
+
+
+@pytest.mark.parametrize(
+    ("exception", "expected_error"),
+    [(IturanApiError, "cannot_connect"), (Exception, "unknown")],
+)
+async def test_errors(
+    hass: HomeAssistant,
+    mock_ituran: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    exception: Exception,
+    expected_error: str,
+) -> None:
+    """Test connection errors during configuration flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    mock_ituran.request_otp.side_effect = exception
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ID_OR_PASSPORT: MOCK_CONFIG_DATA[CONF_ID_OR_PASSPORT],
+            CONF_PHONE_NUMBER: MOCK_CONFIG_DATA[CONF_PHONE_NUMBER],
+        },
+    )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["step_id"] == "user"
+    assert result2["errors"] == {"base": expected_error}
+
+    mock_ituran.request_otp.side_effect = None
+    result3 = await __do_successful_user_step(hass, result2, mock_ituran)
+
+    mock_ituran.authenticate.side_effect = exception
+    result4 = await hass.config_entries.flow.async_configure(
+        result3["flow_id"],
+        user_input={
+            CONF_OTP: "123456",
+        },
+    )
+
+    assert result4["type"] is FlowResultType.FORM
```

Comment:
You don't use old `resultn`s anymore, so you can just overwrite. Not a blocker, but it does look a bit better

Reply:
Done